packages/salesforcedx-vscode-core/src/conflict/timestampConflictDetector.ts

```diff
@@ -51,8 +53,11 @@ export class TimestampConflictDetector {
         component.cacheComponent.fullName
       );
       lastModifiedInCache = cache.getPropertiesForFile(key)?.lastModifiedDate;
-      if (!lastModifiedInCache || lastModifiedInOrg !== lastModifiedInCache) {
-        const differences = diffComponents(component.projectComponent, component.cacheComponent);
+      if (!lastModifiedInCache || lastModifiedInOrg > lastModifiedInCache) {
```

Comment:
This date string comparaison still kind of blows my mind.  According to this [SO](https://stackoverflow.com/questions/14781153/how-to-compare-two-string-dates-in-javascript) this method will work for format YYYY-MM-DD, but it appears to be working for our format as well (at least for the dates we've tested).  I was kind of surprised (but not really) that we don't have a util for working with dates. To avoid blowing up this PR how would you feel about making a private method here to actually compare the dates ala ```
private dateIsGreater(dateStrOne: string, dateStrTwo: string): boolean {
  const dateNumOne = new Date(dateStrOne).getTime();
  const dateNumTwo = new Date(dateStringTwo).getTime();
  return dateNumOne > dateNumTwo;
}
```
working with dates always makes me nervous b/c it's always more complicated then it seems like it should be. 
In the future we should def pull in a module targeting date handling and have a service for handing it.